app/bundles/LeadBundle/Model/LeadModel.php

```diff
@@ -1714,6 +1736,24 @@ public function importLead($fields, $data, $owner = null, $list = null, $tags =
             $merged = false;
         }
 
+        // Extract company data and import separately
+        // Modifies the data array
+        $company                           = null;
+        list($companyFields, $companyData) = $this->companyModel->extractCompanyDataFromImport($fields, $data);
+
+        if (!empty($companyData)) {
+            $companyFields = array_flip($companyFields);
+            $this->companyModel->import($companyFields, $companyData, $owner, $list, $tags, $persist, $eventLog);
+            $companyFields = array_flip($companyFields);
+
+            $companyName    = isset($companyFields['companyname']) ? $companyData[$companyFields['companyname']] : null;
+            $companyCity    = isset($companyFields['companycity']) ? $companyData[$companyFields['companycity']] : null;
+            $companyCountry = isset($companyFields['companycountry']) ? $companyData[$companyFields['companycountry']] : null;
+            $companyState   = isset($companyFields['companystate']) ? $companyData[$companyFields['companystate']] : null;
+
+            $company = $this->companyModel->getRepository()->identifyCompany($companyName, $companyCity, $companyCountry, $companyState);
```

Comment:
At this point I've already imported the company data (more than what is covered by the identifyLeadsCompany method. I need to get the company entity that was just imported so that the lead can be associated with that company. The companyRepo->identifyCompany gives me that functionality.